hashstructure.go

```diff
@@ -242,6 +249,11 @@ func (w *walker) visit(v reflect.Value, opts *visitOpts) (uint64, error) {
 					continue
 				}
 
+				zeroVal := reflect.Zero(reflect.TypeOf(innerV.Interface())).Interface()
+				if w.ignorezerovalue && reflect.DeepEqual(innerV.Interface(), zeroVal) {
```

Comment:
A couple suggestions here:

  * `zeroVal` is calculated every time outside this regardless of `w.ignorezerovalue`. I would recommend calculating it within the conditional since hashstructure is used in some very hot paths (Vault) and if we don't have to calculate the zero that'd be ideal.

  * Instead of `reflect.DeepEqual`, I *think* we can just do `innerV.Interface() == zeroVal` since zero values are always simple values. But that may not be true for maps/slices... I'm not sure. But if it is true its worth changing cause it'll surely be faster than a func call to DeepEqual.